metamist/parser/generic_parser.py

```diff
@@ -198,15 +208,17 @@ def __init__(
         rows: GroupedRow,
         internal_pid: int | None,
         external_pid: str,
-        meta: Dict[str, Any],
-        reported_sex,
-        reported_gender,
-        karyotype,
+        meta: dict[str, Any],
+        reported_sex: int | None,
+        reported_gender: str | None,
+        karyotype: str | None,
+        external_pids: dict[str, str] | None = None,
     ):
         self.rows = rows
 
         self.internal_pid = internal_pid
         self.external_pid = external_pid
+        self.external_pids = external_pids or {}
```

Comment:
Might deserve a comment noting that — unlike other layers — the primary one isn't in `external_pids` in this class.